### PR TITLE
[IS 5.2.0] OIDC Session Management Endpoint Implementation

### PR DESCRIPTION
**Pull request by hasanthiP**
The endpoint which retuens the status of the IS server has been implemented.

### CODE REVIEW

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2005-2013, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thariyarox) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java]: Incorrect License Header. Please use the correct one.

File: components/oauth/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java [hunk 2]
```diff
@@ -70,7 +71,9 @@
     public static String OAUTH_IWA_NTLM_GRANT_ENUM = "IWA_NTLM";
     // OAuth client authenticator properties
     public static final String CLIENT_AUTH_CREDENTIAL_VALIDATION = "StrictClientCredentialValidation";
-
+    //OIDC Session Management Endpoint constants
+    public static final String OP_STATUS_LOGGED="true";
```
Comment (hpmtissera) [inline components/oauth/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java]: format the code

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+
+/**
+ * This class consists OpenID Connect Endpoint which returns the login status of the IS server
+ */
+@Path("/loginstatus")
+public class OIDCSessionMgtEndpoint {
+    static String callBackurl = null;
```
Comment (hpmtissera) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java]: use proper access modifiers

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java [hunk 4]
```diff
@@ -624,7 +625,7 @@ private String doUserAuthz(HttpServletRequest request, String sessionDataKey,
      */
     private OAuth2AuthorizeRespDTO authorize(OAuth2Parameters oauth2Params
             , SessionDataCacheEntry sessionDataCacheEntry) {
-
+        OIDCSessionMgtEndpoint oidcSessionMgtEndpoint=new OIDCSessionMgtEndpoint();
```
Comment (hpmtissera) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java]: not formatted properly

File: components/oauth/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java (hunk 2 above)
Comment (DMHP) [inline components/oauth/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java]: formatted

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java (hunk 3 above)
Comment (DMHP) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java]: Done

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java (hunk 4 above)
Comment (DMHP) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java]: Formatted

**Pull request closed without merging**